Drivers/Inc/CANbus.h

```diff
@@ -4,73 +4,125 @@
 #define CAN_H__
 
 #include "BSP_CAN.h"
+#define CARCAN CAN_1 //convenience aliases for the CANBuses
+#define MOTORCAN CAN_3
 
-typedef enum {
-	MC_BUS = 0x242,
-	VELOCITY = 0x243,
-	MC_PHASE_CURRENT = 0x244,
-	VOLTAGE_VEC = 0x245,
-	CURRENT_VEC = 0x246,
-	BACKEMF = 0x247,
-	TEMPERATURE = 0x24B,
-	ODOMETER_AMPHOURS = 0x24E,
-	ARRAY_CONTACTOR_STATE_CHANGE = 0x24F,
-	CAR_STATE = 0x580,
-	CHARGE_ENABLE = 0x10C,
-	SUPPLEMENTAL_VOLTAGE = 0x10B,
-	STATE_OF_CHARGE = 0x106
+//macros to define canmsg length in LUT
+#define BYTE 1
+#define HALFWORD 2
+#define WORD 4
+#define DOUBLE 8
+
+/**
+ * @brief Struct to use in CAN MSG LUT
+ * @param idxEn Whether or not this message is part of a sequence of messages.
+ * @param size Size of message's data
+ * @param ID The actual CAN ID of this message
+ */
+typedef struct {
+	bool idxEn;
+	uint8_t size;
+	uint32_t ID;
+} CANLUT_T;
+/**
+ * @brief This enum is used to signify the ID of the message you want to send. 
+ * It is used internally to index our lookup table (above) and get message-specific fields.
+ * If changing the order of this enum, make sure to mirror that change in the lookup table, or
+ * else the driver will not work properly.
+ */
+typedef enum { 
+	CHARGE_ENABLE = 0,
+	STATE_OF_CHARGE,
+	SUPPLEMENTAL_VOLTAGE,
+	CAR_STATE,
+	MC_BUS,
+	VELOCITY,
+	MC_PHASE_CURRENT,
+	VOLTAGE_VEC,
+	CURRENT_VEC,
+	BACKEMF,
+	TEMPERATURE,
+	ODOMETER_AMPHOURS,
+	ARRAY_CONTACTOR_STATE_CHANGE,
+	MOTOR_DRIVE,
+	MOTOR_POWER,
+	MOTOR_RESET,
+	MOTOR_STATUS,
+	MOTOR_VELOCITY,
+	NUM_CAN_IDS
 } CANId_t;
 
 
-typedef union {
-	uint8_t b;
-	uint16_t h;
-	uint32_t w;
-	uint64_t d;
-} CANData_t;
+/**
+ * @brief Lookup table to simplify user-defined packet structs. Contains fields that are always the same for every message of a given ID.
+ * Indexed by CANId_t values.
+ * FOR DRIVER USE ONLY.
+ */
+CANLUT_T CANLUT[NUM_CAN_IDS] = {
+	{false, DOUBLE, 0x10C},
+	{false, DOUBLE, 0x106},
+	{false, DOUBLE, 0x10B},
+	{false, DOUBLE, 0x580},
+	{false, DOUBLE, 0x242},
+	{false, DOUBLE, 0x243},
+	{false, DOUBLE, 0x244},
+	{false, DOUBLE, 0x245},
+	{false, DOUBLE, 0x246},
+	{false, DOUBLE, 0x247},
+	{false, DOUBLE, 0x248},
+	{false, DOUBLE, 0x24E},
+	{false, BYTE  , 0x24F},
+	{false, DOUBLE, 0x221},
+	{false, DOUBLE, 0x222},
+	{false, DOUBLE, 0x223},
+	{false, DOUBLE, 0x241},
+	{false, DOUBLE, 0x243},
+};
```

Comment:
This is what is causing your linker errors... Why not just declare this in CANbus.c?